tapchannelmsg/records.go

```diff
@@ -1907,3 +1908,134 @@ func DecodeAuxShutdownMsg(blob tlv.Blob) (*AuxShutdownMsg, error) {
 
 	return &s, nil
 }
+
+// VpktList is a record that represents a list of vPkts.
+type VpktList struct {
+	// Pkts is the list of vPkts.
+	Pkts []*tappsbt.VPacket
+}
+
+// NewVpktList creates a new VpktList record with the given list of vPkts.
+func NewVpktList(pkts []*tappsbt.VPacket) VpktList {
+	return VpktList{
+		Pkts: pkts,
+	}
+}
+
+// Record returns a tlv.Record that represents the VpktList.
+func (v *VpktList) Record() tlv.Record {
+	size := func() uint64 {
+		var (
+			buf     bytes.Buffer
+			scratch [8]byte
+		)
+		err := eVpktList(&buf, v, &scratch)
+		if err != nil {
+			panic(err)
+		}
+
+		return uint64(buf.Len())
+	}
+
+	return tlv.MakeDynamicRecord(0, v, size, eVpktList, dVpktList)
+}
+
+func eVpktList(w io.Writer, val interface{}, buf *[8]byte) error {
+	if v, ok := val.(*VpktList); ok {
+		numPkts := uint64(len(v.Pkts))
+		if err := tlv.WriteVarInt(w, numPkts, buf); err != nil {
+			return err
+		}
+
+		for _, pkt := range v.Pkts {
+			if err := pkt.Serialize(w); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+	return tlv.NewTypeForEncodingErr(val, "*VpktList")
+}
+
+func dVpktList(r io.Reader, val interface{}, buf *[8]byte, _ uint64) error {
+	if typ, ok := val.(*VpktList); ok {
+		numPkts, err := tlv.ReadVarInt(r, buf)
+		if err != nil {
+			return err
+		}
+
+		if numPkts == 0 {
```

Comment:
Do we also need to sanity check the `numPkts` to avoid OOM below?

Reply:
What would we limit it to? FWIW, we're always the ones generating these vPackets. 

I guess we can limit it to the total amount of outputs you can have on a commitment transaction. Will limit it to 1k for some breathing room.